pkg/client/channel/invoke/txnhandler.go

```diff
@@ -63,7 +64,13 @@ func (e *EndorsementHandler) Handle(requestContext *RequestContext, clientContex
 
 	requestContext.Response.Responses = transactionProposalResponses
 	if len(transactionProposalResponses) > 0 {
-		requestContext.Response.Payload = transactionProposalResponses[0].ProposalResponse.GetResponse().Payload
+		responsePayload, err := getResultFromProposalResponse(transactionProposalResponses[0].ProposalResponse)
```

Comment:
Does this need to fallback to `transactionProposalResponses[0].ProposalResponse.GetResponse().Payload` to deal with transaction error responses, or is that handled somewhere else? Passing all the tests so seems this is OK?

Reply:
@troyronda Can you confirm the desired behaviour (see comment above)?